### PR TITLE
Add a session variable to show or hide hidden columns

### LINKED ISSUE #4509: Add a session variable to show or hide hidden columns
Sometimes it is necessary to display the value from the hidden column to help debug, such as determining whether the result of batch deletion of imported data is correct, whether the result of compaction is correct, etc. In the current implementation, this information is completely hidden and cannot be viewed. Therefore, you need to add a session variable `show_hidden_columns` to control whether to show hidden columns and filter hidden column data.
When `show_hidden_columns` is true, functions that get the table's schema such as get_schema will not return hidden columns at the beginning, and the filter condition of `__DELETE_SIGN__=true` will be added to olap sacanNode. When `show_hidden_columns` is false, the opposite is true

### PR DESCRIPTION
**Pull request by yangzhg**
## Proposed changes

Sometimes we need to show hidden columns for debug.
So we need to add a session variable to show or hide hidden columns

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have create an issue on (Fix #4509), and have described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [x If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by chaoyli** [comment]
1. Sometimes is when and where to hidden the columns ?
2. Can be hidden the columns transparent to users. Because that it's only meaningful to doris, not to user.

**Comment by morningman** [comment]
Hi @yangzhg , I think we need a document first to explain the logic of whether the hidden column is displayed or not.
So that it can provide guidance for future development.

**Comment by yangzhg** [comment]
> Hi @yangzhg , I think we need a document first to explain the logic of whether the hidden column is displayed or not.
> So that it can provide guidance for future development.

I have update the issue about the  session variable logic, and add batch delete documents in this pr

File: fe/fe-core/src/main/java/org/apache/doris/common/util/Util.java [hunk 1]
```diff
@@ -477,5 +478,14 @@ public static InputStream getInputStreamFromUrl(String urlStr, String encodedAut
         conn.setReadTimeout(readTimeoutMs);
         return conn.getInputStream();
     }
+
+    public static boolean showHiddenColumns() {
+        try {
+            return ConnectContext.get().getSessionVariable().showHiddenColumns();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/common/util/Util.java:484]: ConnectContext.get() may return null, better check it first, and if it is null, return a default value.

File: fe/fe-core/src/main/java/org/apache/doris/planner/SchemaScanNode.java [hunk 2]
```diff
@@ -91,6 +91,7 @@ protected void toThrift(TPlanNode msg) {
                 msg.schema_scan_node.setDb("SESSION");
             }
         }
+        msg.schema_scan_node.show_hidden_cloumns = ConnectContext.get().getSessionVariable().showHiddenColumns();
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/SchemaScanNode.java:94]: Use `Util. showHiddenColumns()` instead

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**